### PR TITLE
Support rotating SNI certificates in AWS plugin

### PR DESCRIPTION
**Pull request by bobmshannon**
This PR makes the required modifications to support rotating SNI certificates in the AWS plugin.

- Adds a new `replace_sni_certificate` method to the source plugin class.
- Modifies the `deployments` controller (which controls the rotation operations via the source plugins) to be aware of whether there is multiple certificates attached to an endpoint.
- Update the `rotate` CLI sub-command to be aware of whether there is multiple certificates attached to an endpoint.
- Merge the `rotate` and `rotate_region` CLI sub-commands. This was an outstanding TODO, but made sense to do in this PR because it significantly simplifies the changes required.

Closes [EDGE-1365](https://datadoghq.atlassian.net/browse/EDGE-1365).

**Testing**

- Added additional unit test cases to verify functionality of both the source plugin and the rotate CLI sub-command.
- Tested manually end-to-end on sandbox using a test load balancer.

### CODE REVIEW

File: lemur/certificates/cli.py
Comment (ahmad-ddog) [inline lemur/certificates/cli.py:293]: the print and message lines are redundant, maybe just create a message string and print it then assign it to log_data["message"]

File: lemur/certificates/cli.py
Comment (ahmad-ddog) [inline lemur/certificates/cli.py:328]: nit: why change this from "if not endpoint.certificate.replaced"? the original looks more "pythonic"

**Review by ahmad-ddog — DISMISSED** [review]
Code looks LGTM, but can you add more details on the test plan (what steps/commands used, and what results?). This just helps someone like me follow along and gain context, but also serves as future doc for us if we are to make changes on the same file.
Also it's probably a good idea to get a review from someone with more context but this should unblock you for now.

**Review by mitchcail — APPROVED** [review]

**Comment by mitchcail** [comment]
Sorry about the delay on the reviewing. Im taking the morning to look into this!

File: lemur/deployment/service.py
Comment (mitchcail) [inline lemur/deployment/service.py:20]: Im trying to think through if we could have a potential issue in the case that this logic block was hit? 
Lets say we end up in a case where a single certificate is indeed both the primary AND in the list of SNI's. For this thought experiment lets say that `cert1` is currently on the endpoint and we want to rotate in `cert2`. 
In this case our endpoint will return the following values (before rotation)
```
my_endpoint.primary_certificate = cert1
my_endpoint.certificates_assoc = [cert1]
```

Once we hit this logic path it will call `_rotate_primary_certificate` which will set the primary certificate to be `cert2`. So now, if I understand correctly, our state looks like the following

```
my_endpoint.primary_certificate = cert2
my_endpoint.certificates_assoc = [cert1]
```

Is there any issues that could come up with `cert1` still being associated as an sni_certificate after rotation of the primary?

File: lemur/plugins/lemur_aws/plugin.py
Comment (mitchcail) [inline lemur/plugins/lemur_aws/plugin.py:475]: I know there was some conversation in the past about how we should handle removing the old cert. Did we confirm that removing the old certificate immediately after adding the new one is going to be fine. 
My concern here would be that on the AWS side, the upload of the new cert takes longer than removing the old one and then there is a situation where requests will fail for some amount of time.

File: lemur/deployment/service.py
Comment (bobmshannon) [reply lemur/deployment/service.py:20]: This will not be a problem in AWS as `cert2` would be served in that case. Additionally, `cert1` would be rotated on the next automated scheduled rotation attempt.

File: lemur/plugins/lemur_aws/plugin.py
Comment (bobmshannon) [reply lemur/plugins/lemur_aws/plugin.py:475]: See https://datadoghq.atlassian.net/wiki/spaces/NE/pages/2594769333/AWS+load+balancers+SNI+certificate+rotation+and+selection+behavior, this won't be a problem at all.

File: lemur/deployment/service.py
Comment (bobmshannon) [reply lemur/deployment/service.py:20]: That being said we _could_ have Lemur rotate both certificates here if we really wanted to.

Comment (mitchcail) [reply lemur/deployment/service.py:20]: > Additionally, cert1 would be rotated on the next automated scheduled rotation attempt.

This was my assumption but I wanted to confirm. 

> That being said we could have Lemur rotate both certificates here if we really wanted to.

Given your response in the first comment, I don't think  this is needed. Worst case it sounds like the old SNI cert sticks around for another rotation period before being replaced, which we know will be fine.

File: lemur/plugins/lemur_aws/plugin.py
Comment (mitchcail) [reply lemur/plugins/lemur_aws/plugin.py:475]: Nice 👍

File: lemur/deployment/service.py
Comment (bobmshannon) [reply lemur/deployment/service.py:20]: Agreed, if it helps add context by the way, part of my reasoning for initially choosing to only rotate the primary certificate in this case because that's exactly what happens today. Given that I want to contribute this back to the upstream project, maintaining the least surprising behavior is likely to be ideal although what constitutes surprising behavior can be open to interpretation.

File: lemur/plugins/lemur_aws/elb.py
Comment (mitchcail) [inline lemur/plugins/lemur_aws/elb.py:469]: Im wondering if this logic block should be.
```
if e.response["Error"]["Code"] == "LoadBalancerNotFound":
    current_app.logger.warning("Loadbalancer does not exist.")
raise e
```
Just so that we are always raising an exception if this block is hit. Unless there is a reason that we want to continue execution flow if the LB does not exist?

Comment (bobmshannon) [reply lemur/plugins/lemur_aws/elb.py]: Updated

**Review by mitchcail — APPROVED** [review]
All comments were addressed. Happy to smash the approve button on this one 🙌 Can't wait to see this out in production.

**Pull request merged**
